### PR TITLE
texture: add image mask support

### PR DESCRIPTION
**Pull request by Sanheiii**
Masking would not be used in TextureMesh, now I implemented it. 
Masks are sometimes not binarized, so I think we should exclude all values that are not greater than nIgnoreMaskLabel.

### CODE REVIEW

**Comment by Sanheiii** [comment]
Sorry, I found that in some cases it still throws an exception, I will try to fix it.

**Comment by Sanheiii** [comment]
I fixed the above problem and tested it, and it works fine now.

File: libs/MVS/SceneTexture.cpp [hunk 1]
```diff
@@ -535,6 +542,9 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
 				ASSERT((idxFace == NO_ID && depthMap(j,i) == 0) || (idxFace != NO_ID && depthMap(j,i) > 0));
 				if (idxFace == NO_ID)
 					continue;
+				if (!mask.empty() && !mask.isSet(j, i)) {
```
Comment (cdcseacave) [inline libs/MVS/SceneTexture.cpp:545]: this is not ok, there will be cases where the face partially overlaps the invalid region of the mask, so the few pixels that remain valid will assign the view to the mask even though the face contains invalid pixels; pls see PR #968 for a correct implementation; pls build on top of it; if so, pls move the mask estimation in that PR ourside under a command line paramter, to be estimated if the scene does not contain mask files.

Comment (Sanheiii) [reply libs/MVS/SceneTexture.cpp:545]: Ok, I understand the right way to implement it, I will finish it.

**Comment by cdcseacave** [comment]
looking good, great work, I still wait for the #968 PR to be fixed, and I'll let you know when it is merged

**Comment by EliDavis3D** [comment]
It looks like #968 is merged, does that mean this is valid now?

**Comment by cdcseacave** [comment]
@Sanheiii thx for this PR; I modified it a bit though: I removed all mask processing (should be done outside OpenMVS, when the masks are produced, including merging more labels into one, and erosion, etc), and unified the mask loading and usage

**Pull request merged**